### PR TITLE
Updating all components using Vue Class Components.

### PR DESCRIPTION
**Pull request by SolomiiaLomnytska22**
**Refs:**
#2 
**Proposed changes:**
Changing existing components using [Vue Class Component](https://class-component.vuejs.org/) approach.

Tested locally. All existing functionality is supported.

https://github.com/SolomiiaLomnytska22/solomiia-training/assets/136481621/35ac8ca9-de62-4e1f-9dd2-5f2eb79e1d50

### CODE REVIEW

File: app/src/components/common/Button.vue [hunk 1]
```diff
@@ -9,24 +9,29 @@
 </template>
 
 <script>
-export default {
+import Component from 'vue-class-component'
+import Vue from 'vue'
+
+@Component({
   props: {
```
Comment (kaytee-te) [inline app/src/components/common/Button.vue]: Please use Vue Property Decorator (`@Prop`)  for props, link for more info: https://blog.logrocket.com/define-properties-vue-property-decorator-typescript/

File: app/src/components/common/Button.vue [hunk 2]
```diff
@@ -9,24 +9,29 @@
...
     }
-  },
-  methods: {
-    handleClick() {
-      this.$emit('click')
-    }
+  }
+})
+export default class Button extends Vue {
+  handleClick() {
```
Comment (kaytee-te) [inline app/src/components/common/Button.vue:34]: Why are you exporting this method?

Comment (SolomiiaLomnytska22) [reply app/src/components/common/Button.vue:34]: I've used an example from https://class-component.vuejs.org/guide/class-component.html#data. 
Screenshot 2023-11-22 at 11 59 08

**Comment by kaytee-te** [comment]
BTW, it may make sense to create a new branch for each issue.

File: app/src/components/common/Button.vue (hunk 2 above)
Comment (kaytee-te) [reply app/src/components/common/Button.vue:34]: Well, declaring a method means just defining the class method itself, without the `export default...` part. [Here ](https://github.com/SolomiiaLomnytska22/solomiia-training/pull/4/files#diff-06a49d7ac580c3a04f39c4c5a261807f5d1b758de91ffed8a018c548f043759dR43)is a correct way of declaring a class method in your code.

And probably, it makes sense to make the Button component as a class which extends Vue, and as a result, you will be able to use inheritance in the future. Like you did [here](https://github.com/SolomiiaLomnytska22/solomiia-training/pull/4/files#diff-9b85a241aecca141fd84aff388fb9f769364fde6b91ac6c5caacfad9a5c3ea3cR35).

Comment (kaytee-te) [reply app/src/components/common/Button.vue:34]: PS, `@Prop` should be defined in the scope of class.

Comment (SolomiiaLomnytska22) [reply app/src/components/common/Button.vue:34]: In the code snippet that you have commented on I actually exported Button component class 
`export default class Button extends Vue`
and then inside this class I defined handleClick() method. Should I remove this method from Button component class or declare it outside the exported class somehow?

Comment (kaytee-te) [reply app/src/components/common/Button.vue:34]: Yeap, you are right, it was a misunderstanding. Thanks for the explanation.

Let's just update the props for this component.

Comment (SolomiiaLomnytska22) [reply app/src/components/common/Button.vue:34]: Okay, thanks!

File: app/src/components/common/Button.vue (hunk 1 above)
Comment (SolomiiaLomnytska22) [reply app/src/components/common/Button.vue]: Thank you for advice! Defined props using Vue Property Decorator in recent commit.

**Review by kaytee-te — APPROVED** [review]

**Comment by SolomiiaLomnytska22** [comment]
Moved all commits relevant to Issue #2 to another branch [`issue#2-added-class-components`](https://github.com/SolomiiaLomnytska22/solomiia-training/tree/issue%232-added-class-components) and created new pull request here - #5.

**Pull request closed without merging**